Sources/BraintreePayPalNativeCheckout/BTPayPalNativeTokenizationClient.swift

```diff
@@ -33,15 +33,18 @@ class BTPayPalNativeTokenizationClient {
         ) { body, _, error in
             guard let json = body, error == nil else {
                 let underlyingError = error ?? BTPayPalNativeError.invalidJSONResponse
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.failure")
                 completion(.failure(.tokenizationFailed(underlyingError)))
                 return
             }
 
             guard let accountNonce = BTPayPalNativeCheckoutAccountNonce(json: json) else {
+                self.apiClient.sendAnalyticsEvent("ios.paypal-native.tokenize.parsing-result.failure")
```

Comment:
I don't think we need to get this nuanced with the failure analytics. I think `"ios.paypal-native.tokenize.failed"` is sufficient for all cases. Especially since we're already returning specific errors to the merchant for these error use cases - we will be able to learn from their integration which error they hit.

I think at the analytics level, we just want to know if it was success or not. Historically, we have over-complicated our analytics logging resulting in too many events with different names that we lose track of.

Reply:
We also created this [naming guide](https://github.braintreeps.com/team-sdk/sdk-knowledge/blob/main/analytics/events/event-names.md) to try to remedy some of that, and create some uniformity amongst platforms.

Reply:
> I don't think we need to get this nuanced with the failure analytics.

This came up during the merchant call for caching errors and the analytics team we worked with during post mortem recommended us getting more nuanced with the events to better track where we are failing when working with those teams. I don't really have a strong preference one way or the other though. 🤷

Reply:
Ah gotcha! Which post mortem was it for just curious? Which situation caused us to get that feedback?

Reply:
But if that's the feedback we got - makes sense to take it 👍

Reply:
@hollabaq86 - could we get your opinion here? I know you've been doing a handful of analytics work on open dev.

TLDR - do we favor more nuanced analytics events such as having several `ios.paypal-native.tokenize.<SPECIFIC_FAILURE_REASON>.failed` events, or instead favor just one `ios.paypal-native.tokenize.failed` event name that covers all tokenization failures?

Reply:
given that this is one of our larger payment methods, let's go granular and cite the specific failure reason if possible. Thanks for pinging me for my thoughts!

Reply:
👍 Let's do it!